### PR TITLE
feat(`alloy-genesis`): pk support

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

missing pk support.

## Solution

Adds missing private key support to alloy genesis, see https://github.com/foundry-rs/foundry/pull/6970#pullrequestreview-1856796777.

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/genesis/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,8 @@ exclude.workspace = true
 [dependencies]
 alloy-primitives.workspace = true
 alloy-rpc-types.workspace = true
+alloy-signer.workspace = true
+k256.workspace = true
```
Comment (DaniPopes) [inline crates/genesis/Cargo.toml:18]: we should re-export this from alloy-signer actually

File: crates/genesis/src/lib.rs [hunk 2]
```diff
@@ -217,6 +224,55 @@ pub struct GenesisAccount {
...
     )]
     pub storage: Option<HashMap<B256, B256>>,
+    /// The account's private key.
+    #[serde(
+        rename = "secretKey",
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "secret_key"
+    )]
+    pub private_key: Option<LocalWallet>,
```
Comment (DaniPopes) [inline crates/genesis/src/lib.rs:234]: Can we just do B256?

Comment (mattsse) [inline crates/genesis/src/lib.rs:234]: this should mention that this is only for testing

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/genesis/src/lib.rs [hunk 3]
```diff
@@ -217,6 +224,55 @@ pub struct GenesisAccount {
...
+}
+
+/// serde support for `secretKey` in genesis
+pub mod secret_key {
+    use alloy_primitives::Bytes;
+    use alloy_signer::LocalWallet;
+    use k256::{ecdsa::SigningKey, SecretKey};
```
Comment (mattsse) [inline crates/genesis/src/lib.rs:241]: this should not be required here,
the LocalWallet type should have all the necessary functions, if not we should add

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/genesis/src/lib.rs (hunk 2 above)
Comment (Evalir) [reply crates/genesis/src/lib.rs:234]: yeah would prefer that actually, much easier to simplify. wdyt @mattsse

Comment (mattsse) [reply crates/genesis/src/lib.rs:234]: @Evalir I'd actually prefer just B256 as well.

point in my review still stands, we should have functions on LocalWallet that take care of that

File: crates/genesis/Cargo.toml (hunk 1 above)
Comment (Evalir) [reply crates/genesis/Cargo.toml:18]: k256 u mean right?

File: crates/genesis/src/lib.rs (hunk 3 above)
Comment (DaniPopes) [reply crates/genesis/src/lib.rs:241]: Not a great idea to serialize pk by default as raw bytes

**Review by mattsse — APPROVED** [review]
lgtm, lets add an issue for the b256 -> LocalWallet and LocalWallet -> B256 calls?

**Pull request merged**